doc/environment.yml

```diff
@@ -0,0 +1,6 @@
+name: nest-doc
```

Comment:
Since we may (will) have different conda environment files included with NEST (for  working on docs, for running NEST, ...), we should have a more specific file name for this yml file, e.g. `nest-doc-conda-env.yml`.  This would also require corresponding changes in the readme and possibly other places.

Reply:
I have discussed this with @terhorstd and our conclusion was that we'd rather stick with the default name. However, I agree that it could be a bit confusing to have a set of those files in each location where we need them and expect users to infer their meaning based on location, i.e.
* top-level `environment.yml` and `requirements.txt` for runtime requirements
* `extras/environment.yml` and `extras/requirements.txt` for build requirements
* `doc/environment.yml` and `doc/requirements.txt` for documentation requirements
* `testsuite/environment.yml` and `testsuite/requirements.txt` for testing requirements 

Alternatively, we could have
* `extras/conda-environment-nest-{doc,test,build,run}.yml`
* `extras/pyenv-requirements-{doc,test,build,run}.txt`

That said, I'd rather prefer to leave this for discussion/fixing in the context #1612. I've cross-posted this comment there, so we have everything in one place.